src/services/product.service.ts

```diff
@@ -1,39 +1,46 @@
 import { Injectable } from '@angular/core';
-import { of } from 'rxjs';
+import { Observable, of } from 'rxjs';
 
 import { Product } from 'src/interfaces/productInterface';
 
 @Injectable({
     providedIn: 'root',
 })
 export class ProductService {
-    getProducts() {
-        const products: Product[] = [
-            {
-                productName: 'Egg Whites',
-                productDetails: 'These are egg whites',
-                imageURL: 'assets/default.png',
-                productPrice: 1.12,
-                shippingPrice: 0.96,
-                feesPrice: 0.0,
-            },
-            {
-                productName: 'Egg Blacks',
-                productDetails: 'These are egg blacks',
-                imageURL: 'assets/default.png',
-                productPrice: 400.12,
-                shippingPrice: 7000.96,
-                feesPrice: 45096.0,
-            },
-            {
-                productName: 'Mushroom Vase',
-                imageURL: 'assets/default.png',
-                productPrice: 16.09,
-                shippingPrice: 5.0,
-                feesPrice: 4.0,
-            },
-        ];
+    mockProducts: Product[] = [
+        {
+            id: '0',
+            productName: 'Egg Whites',
+            productDetails: 'These are egg whites',
+            imageURL: 'assets/default.png',
+            productPrice: 1.12,
+            shippingPrice: 0.96,
+            feesPrice: 0.0,
+        },
+        {
+            id: '1',
+            productName: 'Egg Blacks',
+            productDetails: 'These are egg blacks',
+            imageURL: 'assets/default.png',
+            productPrice: 400.12,
+            shippingPrice: 7000.96,
+            feesPrice: 45096.0,
+        },
+        {
+            id: '2',
+            productName: 'Mushroom Vase',
+            imageURL: 'assets/default.png',
+            productPrice: 16.09,
+            shippingPrice: 5.0,
+            feesPrice: 4.0,
+        },
+    ];
 
-        return of(products);
+    getProducts(): Observable<Product[]> {
+        return of(this.mockProducts);
+    }
+
+    getProduct(id: string): Observable<Product | undefined> {
```

Comment:
Shouldn't id be of type number? Our ID's will likely only be 1,2,3,4...

Reply:
From what I've read, I think IDs are better off as strings if we don't plan to do any calculations with them. If we for some reason wanted to add other characters into IDs in the future, we could if we start off with strings.